core/src/subgraph/instance_manager.rs

```diff
@@ -179,11 +173,10 @@ impl<S: SubgraphStore> SubgraphInstanceManager<S> {
             logger_factory,
             subgraph_store,
             chains,
-            manager_metrics: Arc::new(SubgraphInstanceManagerMetrics::new(
-                metrics_registry.cheap_clone(),
+            metrics_registry: metrics_registry.cheap_clone(),
+            instances: SharedInstanceKeepAliveMap::new(Arc::new(
+                SubgraphInstanceManagerMetrics::new(metrics_registry),
```

Comment:
This would be easier to read if `SharedInstanceKeepAliveMap::new` took just the `metrics_registry` as an argument and set up the `SubgraphInstanceManagerMetrics` itself. (And feel free to shorten some of those names)